lib/node/runner/checker_message.go

```diff
@@ -175,3 +175,39 @@ func BroadcastTransaction(c common.Checker, args ...interface{}) (err error) {
 
 	return
 }
+
+// BroadcastTransactionFromWatcher is sending tx to one of validators.
+// If all validators returns error, it returns error.
+func BroadcastTransactionFromWatcher(c common.Checker, args ...interface{}) error {
+	checker := c.(*MessageChecker)
+	if checker.Conf.WatcherMode == false {
+		return nil
+	}
+	checker.Log.Debug("transaction from client will be sended")
+
+	cm := checker.Consensus.ConnectionManager()
+	var addrs []string
+
+	for _, a := range cm.AllConnected() {
+		if a != checker.LocalNode.Address() {
+			addrs = append(addrs, a)
+		}
+	}
+
+	if len(addrs) <= 0 {
+		return errors.AllValidatorsNotConnected
+	}
+
+	var err error
+	for _, a := range addrs {
+		client := cm.GetConnection(a)
+		_, err = client.SendTransaction(checker.Transaction)
+		if err == nil {
+			// Broaadcast from watcher is that send one of them using client api successfully.
+			checker.Log.Info("send tx to node", "node", a, "tx", checker.Transaction.GetHash())
+			break
+		}
+		checker.Log.Debug("failure to send tx to node", "node", a, "tx", checker.Transaction.GetHash())
+	}
+	return err
```

Comment:
`err` is derived from the last SendTransaction.
Most of the cases those are the same but there is an opportunity that they differ from each other.

How about change like this?
Select an endpoint randomly from the connected node, send transaction once, and return the err.

Reply:
> Most of the cases those are the same but there is an opportunity that they differ from each other.
I don't understand it.  ㅠㅠ

Reply:
I know this part is naive.  But My intent is closer to  primary and secondary, not load balancing.

Reply:
@anarcher What `kfangw` means is, if there are 3 validators, and error occurs in all 3 validators then first two `err`s are ignored. So he wants to pick only one validator randomly and send to it.
@kfangw Am I right? :)
But I think this `try again` to the other validators model looks good.

One more thing. It's not mandatory, just one more option for readability :)
```
for _, a := range addrs {
	client := cm.GetConnection(a)
	_, err = client.SendTransaction(checker.Transaction)
	if err != nil {
		checker.Log.Debug("failure to send tx to node", "node", a, "err", err, "tx", checker.Transaction.GetHash())
		continue
	}
	// Broaadcast from watcher is that send one of them using client api successfully.
	checker.Log.Info("send tx to node", "node", a, "tx", checker.Transaction.GetHash())
}
```